src/libutil/union-source-accessor.cc

```diff
@@ -0,0 +1,82 @@
+#include "source-accessor.hh"
+
+namespace nix {
+
+struct UnionSourceAccessor : SourceAccessor
```

Comment:
This seems to be not a well-behaved `SourceAccessor`
- a single path can contain simultaneously file contents, a directory listing, and a link target
- a subpath of a file or symlink can exist

These will only be noticeable in corner cases, I presume, but we'll want that fixed.

Reply:
Okay, I've removed the check on the file type. So now the first underlying file that exists determines what operations are allowed.